chart/openfaas/templates/prometheus-cfg.yaml

```diff
@@ -39,6 +39,38 @@ data:
         static_configs:
           - targets: ['localhost:9090']
 
+      # Capture endpoints in the openfaas namespace with a scrape annotation
+      # such as the gateway-provider service.
+      - job_name: 'openfaas-endpoints'
```

Comment:
@LucasRoesler to make this change I went for a combination of a couple of your suggestions.

1) New service for the provider, limited in scope to only a ClusterIP
2) Endpoint rather than Service SD for K8s

Then I restricted this further by only matching where the scrape and port annotations were present.

Reply:
Looks good, I'm glad the research made sense and was helpful